### PR TITLE
Added seeded random number generator so we can do tests

### PR DESCRIPTION
**Pull request by jessberg**

### CODE REVIEW

File: sim/src/edge.rs [hunk 1]
```diff
@@ -20,6 +20,7 @@ pub struct Edge {
     delay: u64,
     id: String,
     neighbors: Vec<String>,
+    seed: Option<u64>,
```
Comment (fruffy) [inline sim/src/edge.rs]: Just generate a seed if it is not provided, then you do not have to use Option

File: sim/src/edge.rs [hunk 2]
```diff
@@ -93,15 +94,14 @@ impl Edge {
...
-                    let idx = rand::thread_rng().gen_range(0, neigh_len);
-                    let mut which_neighbor = self.neighbors[idx].clone();
-                    // Choose a random neighbor to send to, but do not send it back to the one who sent it to you
-                    while which_neighbor == queue_element_to_remove.sender
-                        && self.neighbors.len() > 1
-                    {
-                        let idx = rand::thread_rng().gen_range(0, neigh_len);
-                        which_neighbor = self.neighbors[idx].clone();
+                    let idx;
+                    if self.seed.is_none() {
```
Comment (fruffy) [inline sim/src/edge.rs]: And this will not be necessary, if a seed is always generated.

File: sim/src/main.rs [hunk 3]
```diff
@@ -26,20 +26,28 @@ fn main() {
...
+        )
         .get_matches();
 
     // Set up library access
     let plugin_str = matches.value_of("plugin");
+    let seed = matches.value_of("random_num_seed");
 
     // Create simulator object.
-    let mut simulator: Simulator = Simulator::new();
+    let mut simulator: Simulator = Simulator::new(seed);
```
Comment (fruffy) [inline sim/src/main.rs:52]: If this is None, generate seed and print. With that a run can always be reproduced.

**Review by fruffy — CHANGES_REQUESTED** [review]

File: sim/src/edge.rs (hunk 1 above)
Comment (jessberg) [reply sim/src/edge.rs]: Good idea;  it is done.

File: sim/src/main.rs [hunk 4]
```diff
@@ -26,20 +27,36 @@ fn main() {
...
     // Set up library access
     let plugin_str = matches.value_of("plugin");
+    let seed_arg = matches.value_of("random_num_seed");
+    let seed;
+    if seed_arg.is_none() {
+        let mut rng = rand::thread_rng();
+        seed = rng.gen::<u64>();
+        print!("Using seed {0}\n", seed);
+    } else {
+        seed = seed_arg.unwrap().parse().unwrap();
```
Comment (fruffy) [inline sim/src/main.rs]: Can this parse call be made explicit (u64)? Other than that lgtm.

**Review by fruffy — APPROVED** [review]

**Pull request merged**